### PR TITLE
Rebalances blobbernaut creation

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
rscadd: Overmind-created blobbernauts are now player-controlled if possible.
tweak: Creating a blobbernaut no longer destroys the factory, instead doing heavy damage to it and preventing it from spawning spores for one minute.
tweak: You cannot spawn blobbernauts from overly damaged factories.
/:cl:

Also fixes some runtimes when killing mice.

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/blob_reagents.dm [hunk 1]
```diff
@@ -135,11 +143,13 @@
...
 	reac_volume = ..()
-	M.apply_damage(0.4*reac_volume, TOX)
-	M.hallucination += 0.6*reac_volume
-	M.druggy += 0.6*reac_volume
-	if(M.reagents)
-		M.reagents.add_reagent("spore", 0.2*reac_volume)
+	if(M)
+		M.apply_damage(0.4*reac_volume, TOX)
+	if(M)
+		M.hallucination += 0.6*reac_volume
```
Comment (RemieRichards) [inline code/modules/reagents/chemistry/reagents/blob_reagents.dm]: if the above can destroy the mob, why not do it AFTER this.
surely that would avoid ```if(M) if(M)```

Comment (Anonus) [inline code/modules/reagents/chemistry/reagents/blob_reagents.dm]: me screaming

File: code/modules/reagents/chemistry/reagents/blob_reagents.dm [hunk 2]
```diff
@@ -295,36 +319,38 @@
...
+							X.throw_at(throw_target, throw_range, 1)
+						else
+							X.throw_at(pull, distance, 1)
 					else
-						for(var/i = 0, i < moving_power, i++)
-							sleep(2)
-							if(!step_towards(X, pull))
-								break
+						if(setting_type)
+							for(var/i = 0, i < moving_power, i++)
```
Comment (RemieRichards) [inline code/modules/reagents/chemistry/reagents/blob_reagents.dm]: these loops are bad

```for(var/i in 0 to moving_power-1)```

File: code/modules/reagents/chemistry/reagents/blob_reagents.dm [hunk 3]
```diff
@@ -295,36 +319,38 @@
...
+								sleep(2)
+								if(!step_away(X, pull))
+									break
+						else
+							for(var/i = 0, i < moving_power, i++)
```
Comment (RemieRichards) [inline code/modules/reagents/chemistry/reagents/blob_reagents.dm]: see above

**Comment by GunHog** [comment]
Sexiness! How are the blobbernauts chosen?

**Comment by Anonus** [comment]
From ghosts that aren't AFK, like aliens are. There's no prompt because I imagine it'd feel awful to make a blobbernaut, have it rush off, and then have it gain a key in the middle of like five people and die.

**Comment by Iamgoofball** [comment]
:+1:

**Comment by Zilenan91** [comment]
Nice now blobbernauts are actually worth making

**Comment by Zilenan91** [comment]
Oh hey Blobbernauts can use blob chat right? Also what does the main blob using rally do to player nauts

**Comment by PKPenguin321** [comment]
blobbernauts were already worth making, now they'll be AFK half the time because ghosts in blob rounds always alt+tab lol
do the factories heal themselves passively?

**Comment by Iamgoofball** [comment]
@PKPenguin321 just gotta make it play a "you're an antag now fgt" sound

**Comment by JJRcop** [comment]
@Zilenan91 Apparantly nothing right now, but maybe in the future we could develop a general waypoint system and use it for that.

**Comment by PKPenguin321** [comment]
it would be cool if the blobbernaut was AI controlled but could be taken over at any point by ghosts like swarmer shells. that way it is guaranteed to not be AFK.

**Comment by Zilenan91** [comment]
Well I'd assume that if it can't find any players it's NPC controlled after that

**Comment by JJRcop** [comment]
Yeah they should probs get an attack_ghost for if they default to AI control.

**Comment by Anonus** [comment]
It uses the alien afk bracket, so they shouldn't be afk except in very unlikely cases, such as someone going afk, the blob spawning a blobbernaut while they're still viable, and it selecting them as a candidate.
Giving it an attack_ghost is a solution but I'd rather not, as it'd be unlike most other things anyway.

I can easily add a sound to spawning a blobbernaut.

**Comment by tkdrg** [comment]
Yeah @Anonus just add a sound for feedback, that would be nice.

**Comment by GunHog** [comment]
You could make it do the prompts in chat, like swarmers and such

**Comment by Anonus** [comment]
Added a sound and a brief description of the overmind's chemical(which might be mildly inaccurate* with some chemicals but I'm not fixing it right now)
(*inaccurate being stuff like "produce fragile spores when killed or on expanding", which obviously doesn't apply to blobbernauts)

**Comment by Zilenan91** [comment]
I mean, you could probably make it so sporing pod blobbernauts poop out a spore or two on death but that's probably pretty snowflakey.

**Comment by Anonus** [comment]
It'd be easy to hook it into the blob chem system so not actually that snowflakey, but it'd be something to do later, not in this.

**Pull request merged**